packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.java

```diff
@@ -630,6 +631,7 @@ public void maybeUpdateTypeface() {
   // VisibleForTesting from {@link TextInputEventsTestCase}.
   public void requestFocusFromJS() {
     requestFocusInternal();
+    hasSelectedTextOnFocus = true;
```

Comment:
Why do we avoid the onLayout focus here?

Reply:
The original PR added this behavior to fix `autofocus` and `selectTextOnFocus` not working when used together. Since the focus logic was added to `onLayout` it may trigger in different cases also, you can see that in the first video (`Before` in the table), where the input is focused from JS (with no layout) and then layout change causes the text to be selected even though the focus didn't change.